client/my-sites/stats/features/modules/stats-utm/utm-export-button.jsx

```diff
@@ -26,44 +19,28 @@ function UTMExportButton( { data, fileName } ) {
 		return newData;
 	};
 
-	// Turns the flat array into a CSV string.
-	// Processes the label before export.
+	// Flattens hierarchical UTM data into a simple array for CSV export.
+	// Child items include their parent's label as context.
 	const prepareDataForDownload = ( flatData ) => {
-		const csvData = flatData
-			.map( ( row ) => {
-				// Label should include parent context if present.
-				// ie: "parent label > child label" -- including surrounding quotes.
-				let label = row?.context ? `${ row.context } > ${ row.label }` : row.label;
-				label = label.replace( /"/g, '""' );
-				// Return the label and value.
-				return `"${ label }",${ row.value }`;
-			} )
-			.join( '\n' );
-
-		return csvData;
+		return flatData.map( ( row ) => {
+			let label = row?.context ? `${ row.context } > ${ row.label }` : row.label;
+			label = label.replace( /"/g, '""' ); // Escape double quotes
+			return [ `"${ label }"`, `${ row.value }` ];
+		} );
 	};
 
-	const initiateDownload = ( event ) => {
-		event.preventDefault();
-		recordGoogleEvent( 'Stats', 'CSV Download UTM' );
```

Comment:
[StatsDownloadCSV](https://github.com/Automattic/wp-calypso/blob/16c75d1752f91676098f892446f550b1c42c4567/client/my-sites/stats/stats-download-csv/index.jsx#L60) already handles the event.